### PR TITLE
argocd:make install idempotent

### PR DESCRIPTION
**Pull request by Xieql**
Signed-off-by: Xieql <xieqianglong@huawei.com>

**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:
part of #53 

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
NONE

### CODE REVIEW

File: pkg/plugin/argocd/plugin.go [hunk 1]
```diff
@@ -20,12 +20,14 @@ import (
 	"bytes"
 	"context"
 	"fmt"
+	kubeclient "k8s.io/client-go/kubernetes"
```
Comment (zirain) [inline pkg/plugin/argocd/plugin.go:23]: nit: lint

File: pkg/plugin/argocd/plugin.go [hunk 2]
```diff
@@ -39,10 +41,12 @@ import (
...
 const (
-	component     = "argocd"
-	namespace     = "argocd"
-	argocdService = "argocd-server"
-	passwdSecret  = "argocd-initial-admin-secret"
+	component              = "argocd"
+	namespace              = "argocd"
+	argocdService          = "argocd-server"
+	passwdSecret           = "argocd-initial-admin-secret"
+	namespaceDeleteTimeout = 60 * time.Second
```
Comment (zirain) [inline pkg/plugin/argocd/plugin.go:48]: use timeout/interval in generic.Options

File: pkg/util/policy.go [hunk 3]
```diff
@@ -38,6 +38,20 @@ var ignoredResources = map[schema.GroupVersionKind]struct{}{
...
 
+func GetDefaultClusterPropagationPolicyTypeMeta() metav1.TypeMeta {
+	return metav1.TypeMeta{
+		APIVersion: "policy.karmada.io/v1alpha1",
+		Kind:       "ClusterPropagationPolicy",
+	}
+}
+
+func GetDefaultPropagationPolicyTypeMeta() metav1.TypeMeta {
+	return metav1.TypeMeta{
```
Comment (zirain) [inline pkg/util/policy.go:49]: prefer to remove to pkg/typemeta with var/const
```
PropagationPolicy = metav1.TypeMeta{
		APIVersion: "policy.karmada.io/v1alpha1",
		Kind:       "PropagationPolicy",
	}
```

usage:
```
typemeta.PropagationPolicy 
```

**Comment by zirain** [comment]
BTW, please split this to two commits:
1. fix argcd issue
2. refactor typemeta

File: pkg/util/policy.go (hunk 3 above)
Comment (Xieql) [reply pkg/util/policy.go:49]: May I ask, is using var/const instead of get for performance reasons?

File: pkg/plugin/argocd/plugin.go [hunk 4]
```diff
@@ -149,6 +153,25 @@ func (p *ArgoCDPlugin) argocdManifest() (string, error) {
 	return yaml, nil
 }
 
+func WaitNamespaceDelete(poll, timeout time.Duration, client kubeclient.Interface, namespace string) bool {
+	for start := time.Now(); time.Since(start) < timeout; time.Sleep(poll) {
```
Comment (hzxuzhonghu) [inline pkg/plugin/argocd/plugin.go:157]: make use of wait.PollImmediate

File: pkg/plugin/argocd/plugin.go [hunk 5]
```diff
@@ -149,6 +153,25 @@ func (p *ArgoCDPlugin) argocdManifest() (string, error) {
 	return yaml, nil
 }
 
+func WaitNamespaceDelete(poll, timeout time.Duration, client kubeclient.Interface, namespace string) bool {
```
Comment (hzxuzhonghu) [inline pkg/plugin/argocd/plugin.go:156]: move to pkg/util

Comment (Xieql) [reply pkg/plugin/argocd/plugin.go:156]: ok

**Review by hzxuzhonghu — COMMENTED** [review]
Looks this is not only argocd stuff, can spilit

File: Makefile [hunk 6]
```diff
@@ -101,3 +101,81 @@ gen-check: gen
 .PHONY: serve
 serve:
 	hack/local-docsite-up.sh
+
+# dev test
+.PHONY: start
```
Comment (zirain) [inline Makefile:106]: what's this?

Comment (Xieql) [reply Makefile:106]: ignore it, it is just for test. and I will be deleted after the official submission

**Review by hzxuzhonghu — APPROVED** [review]
/lgtm

**Pull request merged**